Source/Private/TextExperiment/Component/ASTextLayout.m

```diff
@@ -1959,7 +1959,7 @@ - (NSArray *)selectionRectsForRange:(ASTextRange *)range {
   range = [self _correctedRangeWithEdge:range];
   
   BOOL isVertical = _container.verticalForm;
-  NSMutableArray *rects = [NSMutableArray array];
+  NSMutableArray *rects = [[NSMutableArray<NSValue *> alloc] init];
```

Comment:
We may should decide on a standard way to initialize these. E.g. we are now using different ways across the framework:
```
NSMutableArray *rects = [[NSMutableArray<NSValue *> alloc] init];
auto rects = [[NSMutableArray<NSValue *> alloc] init];
NSMutableArray<NSValue *> rects = [[NSMutableArray alloc] init];
```

Reply:
I completely agree. `auto` isn't available in C, so I've been aiming for:

```
C++: auto rects = [[NSMutableArray<NSValue *> alloc] init];
C: NSMutableArray *rects = [[NSMutableArray<NSValue *> alloc] init];
```

We could introduce a macro that evaluates to `__auto_type` like `as_auto` or `autotype`. The only drawback is we can't name the macro `auto` because the old C keyword takes precedence over macros.

I think I'll add `as_auto` in a follow up diff and see what people think.